### PR TITLE
Testing/search recipe usecase

### LINKED ISSUE #33: testing for SearchRecipe use case (and JSONArrayHandler and JSONNullHandler)
right now this has no testing, so the goal is to have as much testing done for this as possible (and reasonable).

### PR DESCRIPTION
**Pull request by seonghyunban**
Created preliminary tests for the recipe search use case.
Have a look and let me know if there's something to be improved, or unclear.

### CODE REVIEW

File: src/main/java/frameworks/api/EdamamRecipeApi.java [hunk 1]
```diff
@@ -19,11 +22,21 @@ public class EdamamRecipeApi implements RecipeAPI {
     private static final String API_ID = System.getenv("EDAMAM_API_ID");
 
     @Override
-    public JSONArray getRecipe(SearchRecipeInputData inputData) {
+    public List<Recipe> getRecipe(SearchRecipeInputData inputData) {
```
Comment (parkhangorodsky) [inline src/main/java/frameworks/api/EdamamRecipeApi.java:25]: nice!

**Review by hhpfiona — APPROVED** [review]
looks good to me

File: pom.xml [hunk 2]
```diff
@@ -14,24 +14,32 @@
...
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
+            <version>5.8.2</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <version>5.12.0</version>
+        </dependency>
```
Comment (parkhangorodsky) [inline pom.xml:42]: a comment explaining how this is different from the other (similar looking) code could be good i think

**Review by parkhangorodsky — APPROVED** [review]
good tests, and good cleanup of the RecipeJSONHandler from the interactor. Left one comment about commenting in the pom.xml file. also, good foundation directory structure for testing (is what it's supposed to be). approved.

**Pull request merged**